S09.03-Exercise-ContentProviderDelete/app/src/androidTest/java/com/example/android/sunshine/SunshineTestCases.java

```diff
@@ -0,0 +1,159 @@
+import android.support.test.runner.AndroidJUnit4;
+import android.widget.ImageButton;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.isClickable;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withContentDescription;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+import static org.hamcrest.core.AllOf.allOf;
+import static org.hamcrest.core.StringContains.containsString;
+//import static android.despresso_samples.recyclerview.TestUtils.withRecyclerView;
+
+@RunWith(AndroidJUnit4.class)
+
+public class SunshineTestCases {
+
+
+    private String navigateDate = "Tomorrow";
+    private String settings = "Settings";
+    private String title = "Sunshine";
+    private String detail = "Details";
+    private String options = "More options";
+    private String location = "Location";
+    private String temp_Units = "Temperature Units";
+    private String cancel_Button = "Cancel";
+    private String metric_Units = "Metric";
+    private String imperial_Units = "Imperial";
+
+
+    @Rule
+    public ActivityTestRule<MainActivity> mActivityRule =
+            new ActivityTestRule(MainActivity.class);
+
+    @Test
+    public void verifyTitle() {
+
+        onView(withText(title)).check(matches(isDisplayed()));  // verify the title of the project
+
+    }
+
+    @Test
+    public void validateRefreshButton() {
+
+        onView(withId(R.id.action_refresh))
+                .check(matches(isDisplayed()))
+                .perform(click())
+                .check(matches(isClickable()));        // Validate if refresh link is clickable
+    }
+
+
+    @Test
+    public void verifyDetailNavigation() {
+
+        // Navigating till Detail Page and Verifying
+
+        onView(allOf(withId(R.id.tv_weather_data), withText(containsString(navigateDate))))  // Will Navigate to detail page of specific date
+                .perform(click());
+        onView(withText(detail))
+                .check(matches(isDisplayed()));
+
+        onView(withText(containsString(navigateDate)))
+                .check(matches(isDisplayed()));               // Confirm if it is navigating to same page by verifying the same text
+
+    }
+
+    @Test
+    public void verifyBackButtonAndNavigate() throws InterruptedException {
+
+
+        verifyDetailNavigation();  // Method called for Navigating
+
+        onView(ViewMatchers.isAssignableFrom(ImageButton.class)) // Click on back link and validating if user is able to navigate again
```

Comment:
Should do a static import of 'isAssignableFrom' so it makes the method a little more readable. Example:
import static android.support.test.espresso.matcher.ViewMatchers.isAssignableFrom;
That way in the method you can call it directly as: 
onView(isAssignableFrom(ImageButton.class))